codice/alliance
# CAL-205 Add support for multicast video streaming

#### What does this PR do?

Add support for multicast video streaming. Improves the video-admin-plugin to display a drop-down select of available network interfaces when a multicast IP address is entered in the URL field.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?

@andrewkfiedler 
@jlcsmith 
@bdeining 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).

@kcwire
@lessarderic

#### How should this be tested?

Configure the video monitor for multicast with the video-admin-plugin. For the IP address in the URL field, use an IP in the range 224.0.0.0 - 239.255.255.255. A drop-down select will appear that allows you to select the network interface that will receive the multicast traffic. The correct interface depends on your environment. 

Use the sample-mpegts-streamgenerator to transmit an mpeg-ts video to the IP address used in the video-admin-plugin. 

Note: Depending on your network configuration/topology, you may encounter a high percentage of packet loss. Most networks to need to be specifically tuned for multicast. For example, I saw 90% packet loss. However, you should still receive enough data to demonstrate the new multicast functionality.

#### Any background context you want to provide?

There isn't a good way to itest the multicast functionality because it relies on the network conditions external to the machine running the test.

Please pay special attention to the JavaScript changes, since I don't have much experience with JS.

#### What are the relevant tickets?

[CAL-205](https://codice.atlassian.net/browse/CAL-205)
#### Screenshots (if appropriate)
#### Checklist:
